pages/verification/index.tsx

```diff
@@ -64,11 +72,12 @@ export default function Verification(props: PageProps) {
                 }, {
                     onSuccess: (data) => {
                         if (data) {
-                            setIsLoading(false);
-                            router.push('/message-sent');
+                            if(replaceRoute) router.replace('/message-sent');
+                            else router.push('/message-sent');
                         }
                         localStorage.removeItem(MessageEnum.tempMessage);
                         localStorage.removeItem(MessageEnum.targetUser);
+                        setIsLoading(false);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes made here are logical and seem to be handling the loading state and route replacement correctly based on the new optional parameters introduced in the `sendAnonymousMessage` function. However, error handling could be improved by adding a catch block to handle any exceptions that might occur during the execution of the function.
```diff
                     onSuccess: (data) => {
                         if (data) {
                             if(replaceRoute) router.replace('/message-sent');
                             else router.push('/message-sent');
                         }
                         localStorage.removeItem(MessageEnum.tempMessage);
                         localStorage.removeItem(MessageEnum.targetUser);
                         setIsLoading(false);
                     },
                     onError: (err) => {
                         setIsLoading(false);
+                    },
+                    catch: (error) => {
+                        console.error('An error occurred:', error);
+                        setIsLoading(false);
                     },
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->